VirtoCommerce.PricingModule.Data/Extensions/IQueryableExtension.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using VirtoCommerce.Domain.Pricing.Model.Search;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.PricingModule.Data.Model;
+using coreModel = VirtoCommerce.Domain.Pricing.Model;
+
+namespace VirtoCommerce.PricingModule.Data.Extensions
+{
+    public static class IQueryableExtension
+    {
+        public static IQueryable<PricelistAssignmentEntity> BuildSearchQuery(this IQueryable<PricelistAssignmentEntity> query, PricelistAssignmentsSearchCriteria criteria)
+        {
+            if (!criteria.PriceListIds.IsNullOrEmpty())
+            {
+                query = query.Where(x => criteria.PriceListIds.Contains(x.PricelistId));
+            }
+
+            if (!string.IsNullOrEmpty(criteria.Keyword))
+            {
+                query = query.Where(x => x.Name.Contains(criteria.Keyword) || x.Description.Contains(criteria.Keyword));
+            }
+
+            var sortInfos = criteria.SortInfos;
+            if (sortInfos.IsNullOrEmpty())
+            {
+                sortInfos = new[] { new SortInfo { SortColumn = ReflectionUtility.GetPropertyName<coreModel.PricelistAssignment>(x => x.Priority) } };
+            }
+
+            query = query.OrderBySortInfos(sortInfos);
+
+            query = query.Skip(criteria.Skip).Take(criteria.Take);
```

Comment:
I think we cannot call Skip and Take here as Count() would change in that case. Need to take it off extension,

Reply:
We do not need to have Count() in this extension, as it leads to query execution.
We need to remove Count() and remove  
query = query.Skip(criteria.Skip).Take(criteria.Take); 
and call them after extension method if it is needed.
We do not need to call count() in DeleteFiltered at all.

Reply:
Made distinct functions while respecting DRY-principle. The idea is to process all criterion fields of supplied parameter. Please correct me if I'm wrong.

Reply:
I think we can leave only one method, which would be without Count() and Skip.Take.
public static IQueryable BuildSearchQuery(this IQueryable query, PricelistAssignmentsSearchCriteria criteria)
In our case i do not see Count and Skip.Take part multiple usage. So adding multiple methods just increase complexity here without profit.

Reply:
For clarity we can call this method 
ApplyFilteringConditions
to explicitly tell that we just applying filtering conditions to query, not executing it.

Reply:
> I think we can leave only one method, which would be without Count() and Skip.Take.
> public static IQueryable BuildSearchQuery(this IQueryable query, PricelistAssignmentsSearchCriteria criteria)
> In our case i do not see Count and Skip.Take part multiple usage. So adding multiple methods just increase complexity here without profit.

I guess, Skip/Take values being a part of supplied PricelistAssignmentsSearchCriteria parameter could be processed by default too. In case we want to remove any executions from IQueryable<> extension we could make Skip/Take part optional. Trying to make extension fully responsible for criterion processing. Please check if it does make sense in latest commit.

Reply:
You could leave ApplySkipTake extension method, but it should not be called in BuildSearchQuery and there shouldn't be such option.
ApplySkipTake call not needed for getting ids in DeleteFiltered, as we want to get all Ids without batches.

Reply:
Shouldn't BuildSearchQuery fully process supplied PricelistAssignmentsSearchCriteria argument? It seems to me, Skip/Take values contribute to the criteria as well as Keyword, PriceListIds... do.

> ApplySkipTake call not needed for getting ids in DeleteFiltered, as we want to get all Ids without batches.
BuildSearchQuery is being called with 'false' flag especially to execute Count( ) before Skip/Take pagination. Alternatively, we could name it BuildSearchQueryNoSkipTake() or BuildSearchQueryWithoutPagination()

Reply:
Let's call method ApplyFilteringWithoutPagination and remove ApplySkipTake from extension at all, as it is just used once in search service.

Reply:
Fixed filtering in Delete-all funciton, made extension method narrower in usage scope